include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -5731,53 +5749,107 @@ namespace internal
+              const Number *in_  = in + di * dofs_per_component;
+              Number *      out_ = out + di * dofs_per_component;
+              evaluator.template hessians<0, true, false>(in_, out_);
+              if (dim > 1)
+                evaluator.template hessians<1, true, false>(out_, out_);
+              if (dim > 2)
+                evaluator.template hessians<2, true, false>(out_, out_);
+            }
+          if (dyadic_coefficients)
+            {
+              const auto n_coeff_components =
+                n_desired_components * n_desired_components;
+              if (n_desired_components == dim)
+                {
+                  for (unsigned int q = 0; q < dofs_per_component; ++q)
+                    vmult<dim>(&inv_coefficient[q * n_coeff_components],
+                               &in[q],
+                               &out[q],
+                               dofs_per_component);
+                }
+              else
+                {
+                  for (unsigned int q = 0; q < dofs_per_component; ++q)
+                    vmult<-1>(&inv_coefficient[q * n_coeff_components],
+                              &in[q],
+                              &out[q],
+                              dofs_per_component,
+                              n_desired_components);
+                }
+            }
+          else
+            for (unsigned int q = 0; q < dofs_per_component; ++q)
+              out[q] *= inv_coefficient[q];
 
+          for (unsigned int di = 0; di < n_comp_inner; ++di)
+            {
+              Number *out_ = out + di * dofs_per_component;
+              if (dim > 2)
+                evaluator.template hessians<2, false, false>(out_, out_);
+              if (dim > 1)
+                evaluator.template hessians<1, false, false>(out_, out_);
+              evaluator.template hessians<0, false, false>(out_, out_);
+            }
+
+          in += dofs_per_component;
+          out += dofs_per_component;
           inv_coefficient += shift_coefficient;
         }
+
       return false;
     }
 
-    /**
-     * Version for degree = -1
-     */
-    template <int fe_degree, int = 0>
-    static bool
-    run(const unsigned int,
-        const AlignedVector<Number> &,
-        const AlignedVector<Number> &,
-        const Number *,
-        Number *,
-        std::enable_if_t<fe_degree == -1> * = nullptr)
+  private:
+    template <int n_components>
+    static inline void
+    vmult(const Number *     inverse_coefficients,
+          const Number *     src,
+          Number *           dst,
+          const unsigned int dofs_per_component,
+          const unsigned int n_given_components = 0)
     {
-      static_assert(fe_degree == -1, "Only usable for degree -1");
-      Assert(false, ExcNotImplemented());
-      return false;
+      const unsigned int n_desired_components =
+        (n_components > -1) ? n_components : n_given_components;
+
+      std::array<Number, dim + 2> tmp;
```

Comment:
```suggestion
      std::array<Number, dim + 2> tmp = {};
```

Reply:
I'm looking into it.

Reply:
I wasn't able to reproduce the error. I can look into it in more detail if your suggestion does not work.